### PR TITLE
Refactor non-config aspects out of PerformanceConfig

### LINKED ISSUE #361: Performance config makes performance collector class
`PerformanceConfig` should build a `PerformanceCollector` class, where it contains timer, info per step, and profiler. And `PerformanceCollector` is not an attribute to `PerformanceConfig`, could be either a property or off a build method. This will also mitigate having to do things like `if self.performance_mode` and 

```
        # TODO: these attributes are popped because they're not really config,
        # we should move them off of config classes and onto non-config classes.
        config_dict["performance_config"].pop("times_per_step", None)
        config_dict["performance_config"].pop("hits_per_step", None)
        config_dict["performance_config"].pop("timestep_timer", None)
        config_dict["performance_config"].pop("total_timer", None)
```

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

Currently PerformanceConfig has attributes that aren't representable by or loaded from yaml, which causes issues when serializing it back to disk. This PR separates those concerns into a PerformanceCollector class.

Resolves #361

## Code changes:

- Added PerformanceCollector, NullPerformanceCollector, and AbstractPerformanceCollector
- renamed "performance_mode" setting in PerformanceConfig to "collect_performance"
- removed runtime values from PerformanceConfig

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://drive.google.com/file/d/1R0nqOxfYnzaSdoYdt8yjx5J482ETI2Ft/view?usp=sharing).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Comment by mcgibbon** [comment]
launch jenkins

File: driver/pace/driver/performance/collector.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            self.times_per_step,
+            self.experiment_name,
+            dt_atmos,
+        )
+
+
+class NullPerformanceCollector:
+    def __init__(self):
+        self.total_timer = pace.util.Timer()
+        self.timestep_timer = pace.util.Timer()
```
Comment (elynnwu) [inline driver/pace/driver/performance/collector.py:87]:
```suggestion
        self.total_timer = pace.util.NullTimer()
        self.timestep_timer = pace.util.NullTimer()
```

Comment (elynnwu) [reply driver/pace/driver/performance/collector.py:87]: I think technically it doesn't matter, but might be more precise.

**Review by elynnwu — APPROVED** [review]

File: driver/pace/driver/performance/collector.py (hunk 1 above)
Comment (mcgibbon) [reply driver/pace/driver/performance/collector.py]: Done.

**Pull request merged**